app/components/publications/batch-documents-publication-modal.js

```diff
@@ -72,19 +77,33 @@ export default class PublicationsBatchDocumentsPublicationModalComponent extends
 
   @task
   *saveNewPublication(publicationProperties) {
-    const publicationFlow = yield this.publicationService.createNewPublicationFromMinisterialCouncil(publicationProperties, {
-      case: this.case,
-      agendaItemTreatment: this.agendaItemTreatment,
-      mandatees: this.mandatees,
-    });
+    const publicationFlow =
+      yield this.publicationService.createNewPublicationFromMinisterialCouncil(
+        publicationProperties,
+        {
+          case: this.case,
+          agendaItemTreatment: this.agendaItemTreatment,
+          mandatees: this.mandatees,
+          regulationType: yield this.getRegulationTypeThroughReferenceDocument(
+            this.referenceDocument
+          ),
+        }
+      );
     this.referenceDocument.publicationFlow = publicationFlow;
     yield this.referenceDocument.save();
+
     this.referenceDocument = null;
     this.isOpenNewPublicationModal = false;
   }
 
   @action
   async linkPublicationFlow(piece, publicationFlow) {
+    const regulationType = await publicationFlow.regulationType;
+    if (!regulationType) {
+      publicationFlow.regulationType =
+        await this.getRegulationTypeThroughReferenceDocument(piece);
+      await publicationFlow.save();
```

Comment:
Hier kunnen we nog extra check toevoegen om de save enkel te doen als `getRegulationTypeThroughReferenceDocument` effectief iets teruggeeft.